### PR TITLE
Fixup two remote test problems

### PR DESCRIPTION
**Pull request by cliping**
Fix 'ShellTimeoutError' for virsh_start and virsh_dommemstat.

Signed-off-by: Liping Cheng <lcheng@redhat.com>

### CODE REVIEW

**Comment by cliping** [comment]
Test result:
#avocado run --vt-type libvirt --vt-machine-type q35 virsh.start.status_error_no.remote_start --vt-connect-uri qemu:///system
JOB ID     : 7bf5f49402543829084bc63e9e09df6306f8bd9b
JOB LOG    : /root/avocado/job-results/job-2019-08-01T04.57-7bf5f49/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.start.status_error_no.remote_start: PASS (7.30 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 8.96 s

File: libvirt/tests/src/virsh_cmd/domain/virsh_start.py [hunk 1]
```diff
@@ -153,6 +148,8 @@ def run(test, params, env):
 
         elif pre_operation == "rename":
             libvirt_xml.VMXML.vm_rename(vm, backup_name)
+        elif pre_operation == "remote":
+            virsh.destroy(vm_ref, ignore_status=True, debug=True, uri=remote_uri)
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/domain/virsh_start.py]: As ignore_status=True, we'd better check the command execution result.

File: libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py [hunk 2]
```diff
@@ -46,27 +43,24 @@ def run(test, params, env):
...
-            command = "virsh -c %s dommemstat %s %s" % (remote_uri, vm_name,
-                                                        extra)
-            status = session.cmd_status(command, internal_timeout=5)
-            session.close()
-        except process.CmdError:
-            status = 1
+    try:
+        if vm_ref != "remote":
+            status = virsh.dommemstat(vm_ref, extra, ignore_status=True,
+                                      debug=True).exit_status
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py]: As ignore_status=True, we'd better check the command execution result.

File: libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py [hunk 3]
```diff
@@ -46,27 +43,24 @@ def run(test, params, env):
...
+        else:
+            remote_ip = params.get("remote_ip", "REMOTE.EXAMPLE.COM")
+            remote_pwd = params.get("remote_pwd", None)
+            remote_user = params.get("remote_user", "root")
+            if remote_ip.count("EXAMPLE.COM"):
+                test.cancel("remote ip parameters not set.")
+            ssh_key.setup_ssh_key(remote_ip, remote_user, remote_pwd)
+            remote_uri = "qemu+ssh://%s/system" % remote_ip
+            virsh.start(vm_name, ignore_status=True, debug=True, uri=remote_uri)
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py]: As ignore_status=True, we'd better check the command execution result.

File: libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py [hunk 4]
```diff
@@ -46,27 +43,24 @@ def run(test, params, env):
...
+            status = virsh.dommemstat(vm_name, ignore_status=True,
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py]: As ignore_status=True, we'd better check the command execution result.

File: libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py [hunk 5]
```diff
@@ -46,27 +43,24 @@ def run(test, params, env):
...
+                                      debug=True, uri=remote_uri).exit_status
+    finally:
+        if vm_ref == "remote":
+            virsh.destroy(vm_name, ignore_status=True, debug=True, uri=remote_uri)
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py]: As ignore_status=True, we'd better check the command execution result.

Comment (cliping) [reply libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py]: Change ignore_status to False.

File: libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py (hunk 3 above)
Comment (cliping) [reply libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py]: Change ignore_status to False.

File: libvirt/tests/src/virsh_cmd/domain/virsh_start.py (hunk 1 above)
Comment (cliping) [reply libvirt/tests/src/virsh_cmd/domain/virsh_start.py]: Change ignore_status to False.

File: libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py (hunk 4 above)
Comment (cliping) [reply libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py]: Check status on line 71.

File: libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py (hunk 2 above)
Comment (cliping) [reply libvirt/tests/src/virsh_cmd/monitor/virsh_dommemstat.py]: Check status on line 71.

**Pull request merged**